### PR TITLE
Fix graph.metrics sync after adding a new query stage

### LINKED ISSUE #44668: Visualization error with a custom column
### Describe the bug

Adding a custom column breaks visualization

Screenshot 2024-06-25 at 5 04 50 AM
Screenshot 2024-06-25 at 5 05 21 AM

### To Reproduce

1. Start with https://stats.metabase.com/question/18346-top-5-states
2. Add a custom column `concat([Count],"")` and call it `Count 2`
3. Sort by `Count 2` asc
4. Click on Visualize

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
stats
```

### Severity

P1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kulyk**
Closes #44668

When a new query stage is added, we lose the `source` info for columns from previous stages, and they're all coming with `source: "fields".` `syncGraphMetricSettings` expect metric columns to always have `source: "aggregation"`, so it drops these columns even when they're available. Fixed by removing the `source` check

### CODE REVIEW

File: frontend/src/metabase/visualizations/lib/sync-settings.ts [hunk 1]
```diff
@@ -99,11 +99,7 @@ function syncGraphMetricSettings(
     return settings;
   }
 
-  const metricColumnNames = new Set(
-    cols
-      .filter(column => column.source === "aggregation")
-      .map(column => column.name),
-  );
+  const metricColumnNames = new Set(cols.map(column => column.name));
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/lib/sync-settings.ts:102]: I think this is incorrect and will auto-add irrelevant columns.

**Review by ranquild — COMMENTED** [review]
Inline

File: frontend/src/metabase/visualizations/lib/sync-settings.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/visualizations/lib/sync-settings.ts:102]: I'm thinking about something like this:
```
// keep metricColumnNames unchanged
const allColumnNames = new Set(cols.map(column => column.name));

// change removedMetricColumnNames
const removedMetricColumnNames = new Set(
    [...prevMetricColumnNames].filter(name => !allColumnNames.has(name)),
  );
```

Comment (kulyk) [reply frontend/src/metabase/visualizations/lib/sync-settings.ts:102]: @ranquild thanks! I'll investigate

**Review by ranquild — APPROVED** [review]

**Pull request merged**